### DISCUSSION TITLE
Dynamic topic modeling - 'float' object has no attribute 'left' issues,

### DISCUSSION THREAD

**Discussion by hingging** [MaartenGr/BERTopic#874] [Q&A]
Hi maaten, 
first of all, Thanks for your amazing work!! 

now i'm currently trying to run dynamic topic modeling with the data i have, and I got 
'float' object has no attribute 'left' issues, 

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-78-63e2cccc1345>](https://localhost:8080/#) in <module>
----> 1 topics_over_time = topic_model.topics_over_time(docs = abstract, 
      2                                                 timestamps=timestamps,
      3                                                 global_tuning=True,
      4                                                 evolution_tuning=True,
      5                                                 nr_bins=20)

5 frames
[/usr/local/lib/python3.8/dist-packages/bertopic/_bertopic.py](https://localhost:8080/#) in <lambda>(row)
    664         if nr_bins:
    665             documents["Bins"] = pd.cut(documents.Timestamps, bins=nr_bins)
--> 666             documents["Timestamps"] = documents.apply(lambda row: row.Bins.left, 1)
    667 
    668         # Sort documents in chronological order

AttributeError: 'float' object has no attribute 'left'

----------    
I guess it is data type issues, can you please help me with this..? 

Thanks heaps in advance@ 2

**Comment by MaartenGr** [comment]
This might have something to do with the `timestamps` variable. Are there at least 20 different values in there? And are they void of NaN values?
